platform/android/export/gradle_export_util.h

```diff
@@ -242,4 +254,190 @@ String _get_application_tag(const Ref<EditorExportPreset> &p_preset, const Strin
+	print_line(asset_path_to);
+	print_line(asset_path_from);
+	da->make_dir_recursive(asset_path_to);
+	//	Error err = da->copy(asset_path_from, asset_path_to);
+	Error err = da->rename(asset_path_from, asset_path_to);
+	if (err != OK) {
+		print_line("could not rename");
+		return err;
+	}
+	print_line("dealing with pack files now");
+	err = _deal_with_pack_files(packInfo.filepath, packInfo);
+
+	return err;
+}
+
+Error _create_asset_pack_folder(const AssetPackInfo &packInfo) {
+	String packName = packInfo.packName;
+	String deliveryType = "install-time"; //default
+	if (packInfo.deliveryMode == INSTALL_TIME) {
+		deliveryType = "install-time";
+	}
+	if (packInfo.deliveryMode == FAST_FOLLOW) {
+		deliveryType = "fast-follow";
+	}
+	if (packInfo.deliveryMode == ON_DEMAND) {
+		deliveryType = "on-demand";
+	}
+	String build_gradle_string = vformat(
+			"apply plugin: 'com.android.asset-pack'\n\n"
+			"assetPack {\n"
+			"    packName = \"%s\"\n"
+			"    dynamicDelivery {\n"
+			"        deliveryType = \"%s\"\n"
+			"    }\n"
+			"}\n",
+			packName, deliveryType);
+	String build_gradle_path = vformat("res://android/build/%s/build.gradle", packName);
+	store_string_at_path(build_gradle_path, build_gradle_string);
+	Error err = _copy_asset_directory_files(packInfo);
+	return err;
+}
+
+Error _read_asset_pack_config_file(const String config_filepath, Vector<AssetPackInfo> &r_asset_pack_info) {
+	//TODO: read asset_pack file line by line and populate r_asset_pack_info
+	FileAccess *f = FileAccess::open(config_filepath, FileAccess::READ);
+	if (f == NULL) {
+		return ERR_DOES_NOT_EXIST;
+	}
+	while (!f->eof_reached()) {
+		AssetPackInfo temp_info;
+		String path = f->get_line().strip_edges();
+		int mode = f->get_line().strip_edges().get(0) - 48; //48 = Ascii 0
+		String name = f->get_line().strip_edges();
+		temp_info.filepath = path;
+		temp_info.deliveryMode = (Asset_Pack_Delivery_Mode)mode;
+		temp_info.packName = name;
+		r_asset_pack_info.push_back(temp_info);
+	}
+	return OK;
+}
+
+Error _handle_asset_packs(const String config_filepath) {
+	//TODO: add some checks here to make sure everything is alright
+	Vector<AssetPackInfo> asset_pack_info;
+	_read_asset_pack_config_file(config_filepath, asset_pack_info);
+	for (int i = 0; i < asset_pack_info.size(); i++) {
+		_create_asset_pack_folder(asset_pack_info.get(i));
+	}
+	_update_root_project_with_asset_pack_info(asset_pack_info);
+	return OK;
+}
+
+Error delete_asset_folders() {
+	print_line("deleting asset folders");
+	DirAccessRef da = DirAccess::open("res://android/build");
+	if (!da) {
+		return ERR_CANT_OPEN;
+	}
+	da->list_dir_begin();
+	String keep[] = { ".", "..", "res", "AndroidManifest.xml", "config.gradle",
+		"libs", "gradle", "gradlew", ".gdignore", "build.gradle",
+		"gradle.properties", "gradlew.bat", "settings.gradle", "src" };
+	while (true) {
+		String file = da->get_next();
+		if (file == "") {
+			break;
+		}
+		bool need_to_delete = true;
+		for (int i = 0; i < 14; i++) {
```

Comment:
Maybe use a vector or something so you don't need to hardcode "14". Better yet something like a hashset with O(1) lookup.